hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/HoodieSparkIndexClient.java

```diff
@@ -86,9 +92,45 @@ public static HoodieSparkIndexClient getInstance(SparkSession sparkSession) {
 
   @Override
   public void create(HoodieTableMetaClient metaClient, String userIndexName, String indexType, Map<String, Map<String, String>> columns, Map<String, String> options) throws Exception {
+    if (indexType.equals(PARTITION_NAME_SECONDARY_INDEX) || indexType.equals(PARTITION_NAME_BLOOM_FILTERS)
+        || indexType.equals(PARTITION_NAME_COLUMN_STATS)) {
+      createGenericIndex(metaClient, userIndexName, indexType, columns, options);
+    } else {
+      createRecordIndex(metaClient, userIndexName, indexType);
+    }
+  }
+
+  private void createRecordIndex(HoodieTableMetaClient metaClient, String userIndexName, String indexType) {
```

Comment:
where are we validting that the column(s) provided in the create index command are all record key fields?